app/models/product.rb

```diff
@@ -38,10 +38,10 @@ class Product < ActiveRecord::Base
     end
   end
 
-  scope :active,             conditions: { is_archived: false, is_hidden: false }
-  scope :active_plus_hidden, conditions: { is_archived: false }
-  scope :archived,           conditions: { is_archived: true }
-  scope :not_archived,       conditions: { is_archived: false }
+  scope :active, -> { where(is_archived: false, is_hidden: false) }
+  scope :active_plus_hidden, -> { where(is_archived: false) } # TODO: is this scope named correctly?
```

Comment:
There's probably a better name for it, but `active` and `hidden` are two separate concepts. I think `active` may actually be the incorrectly named one.